### PR TITLE
feat: batch search results filters

### PR DESCRIPTION
**Pull request by caro3801**
In batch search results, it adds the possibility to filter the table by content types and to exclude selectedQueries from the results.

image

Before merging : 

- [x] merge backend https://github.com/ICIJ/datashare/pull/1133

Add:
- Filtering by contentTypes directly from the table 
- Filtering by queries directly from the table 
- Display active filters (number of queries for selected queries and one by content type); selected queries badge is crossed when the excludeQueries flag is true
- Clear filters button
- The table filtering is driven by the URL params according to this parameters:
* page : the current page starting by one
* sort: the field to sortBy
* order: the sort order (asc or desc)
* queries: the selected queries to filter by
* contentTypes: the selected content types to filter by
* queriesExcluded: the flag to invert the queries filters (exclude selected queries == invert selection)

The table is paginated according to the new pagination described in https://github.com/ICIJ/datashare/pull/1133 . This pagination format has also been added to batch search table and user history.

### CODE REVIEW

File: src/components/AppliedSearchFiltersItem.vue [hunk 1]
```diff
@@ -98,7 +102,10 @@ export default {
       if (this.readOnly) {
         return
       }
-
+      if (!this.isSearchFilter) {
```
Comment (caro3801) [inline src/components/AppliedSearchFiltersItem.vue:105]: rename updateSearchStore

File: src/components/BatchSearchResultsFilters.vue [hunk 2]
```diff
@@ -91,13 +106,19 @@ export default {
...
+      default: () => []
     }
   },
   data() {
     return {
       queriesFilter: null,
-      sortField: 'count',
-      sortFields: ['default', 'count']
+      sortFields: ['default', 'count'],
+      filterQueries: map(this.selectedQueries, (q) => ({
```
Comment (pirhoo) [inline src/components/BatchSearchResultsFilters.vue:119]: Can be replaced by

```
filterQueries: map(this.selectedQueries, (label) => ({ label }))
```

File: src/components/BatchSearchResultsTable.vue [hunk 3]
```diff
@@ -0,0 +1,440 @@
...
+        <template #cell(contentLength)="{ item }">
+          <span class="text-nowrap">{{ getDocumentSize(item.contentLength) }}</span>
+        </template>
+        <template #cell(empty)>
+          <div class="text-center">
+            {{ $t('batchSearchResults.empty') }}
+          </div>
+        </template>
+      </b-table>
+      <!--      </v-wait>-->
```
Comment (caro3801) [inline src/components/BatchSearchResultsTable.vue:95]: remove me

File: src/components/BatchSearchResultsTable.vue [hunk 4]
```diff
@@ -0,0 +1,440 @@
...
+    linkGen(page) {
+      return this.generateLinkToBatchSearchResults({ page })
+    },
+    updateRoute(query) {
+      const to = this.generateLinkToBatchSearchResults(query)
+      if (!isEqual(to.query, this.$route.query)) {
+        return this.$router.push(to)
+      }
+    },
+    generateLinkToBatchSearchResults({
```
Comment (caro3801) [inline src/components/BatchSearchResultsTable.vue:347]: change my name

File: src/components/BatchSearchResultsTable.vue [hunk 5]
```diff
@@ -0,0 +1,440 @@
...
+      this.$store.dispatch('batchSearch/updateBatchSearch', { batchId: this.batchSearch.uuid, published })
+    },
+    openDocumentModal(docIndex) {
+      this.$emit('show-document-modal', docIndex)
+    },
+    localeLongDate(date) {
+      return humanLongDate(date, this.$i18n.locale)
+    },
+    localeShortDate(date) {
+      return moment(date).isValid() ? humanShortDate(date, this.$i18n.locale) : ''
```
Comment (caro3801) [inline src/components/BatchSearchResultsTable.vue:400]: remove moment

**Pull request merged**